src/app/modules/main/wallet_section/transactions/module.nim

```diff
@@ -114,8 +114,8 @@ method viewDidLoad*(self: Module) =
   self.setEnabledChainIds()
   self.setPendingTx()
 
-method switchAccount*(self: Module, accountIndex: int) =
-  let walletAccount = self.controller.getWalletAccount(accountIndex)
+method filterChanged*(self: Module, addresses: seq[string], chainIds: seq[int]) =
+  let walletAccount = self.controller.getWalletAccountByAddress(addresses[0])
```

Comment:
@alaibe in many places here we're accessing an address by index `0` directly (hope that will change when you change the `addresses*: seq[string]` to `address*: string`), but even after that change for the security reason we should chek 
```
if walletAccount.isNil: 
  error "no known wallet account for the provided address", address=address
  return
```

Reply:
The opposite, I will do: `getWalletAccountByAddresses(addresses)`